AMM/contracts/exchange/tests/artifacts/malicious_implementation/src/main.sw

```diff
@@ -0,0 +1,98 @@
+contract;
+
+use libraries::{
+    data_structures::{
+        PoolInfo,
+        PreviewAddLiquidityInfo,
+        PreviewSwapInfo,
+        RemoveLiquidityInfo,
+    },
+    Exchange,
+};
+use std::constants::BASE_ASSET_ID;
+
+storage {
+    pair: Option<(ContractId, ContractId)> = Option::None(),
```

Comment:
The `None()` works? I believe this errors in Rust because `None` does not take any arguments.
cc @mohammadfawaz

Reply:
Ugh yeah that's weird. I'm not surprised it works given that the variant looks like `None: ()`, but it is weird.

Reply:
Nice catch, this contract is not used at all there are no tests that caught this. Thanks.